nexusdialog/src/main/java/com/github/dkharrat/nexusdialog/controllers/LabeledFieldController.java

```diff
@@ -56,7 +73,26 @@ public String getLabel() {
      *                  field can be empty.
      */
     public void setIsRequired(boolean required) {
-        this.required = required;
+        if (! required) {
+            // Multiple instances of the validator may be present.
+            Iterator<InputValidator> iterator = validators.iterator();
+            while (iterator.hasNext()){
+                if(iterator.next() instanceof RequiredFieldValidator) {
+                    iterator.remove();
+                }
+            }
+        } else if (! isRequired()) {
+            validators.add(new RequiredFieldValidator());
```

Comment:
This last method won't work in some specific use cases, e.g.:
1. The user provides a set of validators containing a `new RequiredFieldValidator()`,
2. The user uses the method `LabeledFieldController#setRequired(...)`

`setRequired` will use the protected instance of `RequireFieldValidator` which will be different from the one added by the user. We can still have 2 instances of the `RequireFieldValidator` in the set of `InputValidator`.

It is to be noted that the two other methods will not require an iteration anymore to remove the  `RequireFieldValidator` field from the set.